src/controllers/auth.controller.js

```diff
@@ -14,10 +14,11 @@ const userController = {
         req.body.password,
         req.body.phoneNumber,
         req.body.username,
+        'citizen',
       );
-      users.push(user);
+      await db.insertIntoUser(user);
       return returnMessage(res, 201, 'User created successfully', {
-        userId: user.userId,
+        userId: parseInt(user.userId, 0),
```

Comment:
Hey @shaazk, you are returning a response to a user, before you are sure if data you wanted to record in the database is recorded. So, ensure that you assign `await db.insertIntoUser(user)` to a variable, and then, in the response, you return to a user, use data from the database, not the ones from the user model.